unity_py/resources/collection.py

```diff
@@ -72,13 +81,20 @@ def to_stac(collection, data_dir):
                 "created": dataset.data_create_time if dataset.data_create_time!= None else updated,
                 "updated": updated
             },
-        )
-        catalog.add_item(item)
-        for df in dataset.datafiles:
-            item.add_asset(
-                # key="data", asset=pystac.Asset(href=f,title="Main Data File", media_type=pystac.MediaType.HDF5)
-                key=df.type, asset=Asset(href=df.location,title="{} file".format(df.type))
-                )
+
+            )
+            item.properties.update(dataset.properties)
+            catalog.add_item(item)
+
+            for df in dataset.datafiles:
+                if(Collection.is_uri(df.location)):
```

Comment:
Use https://validators.readthedocs.io/en/latest/#module-validators.url instead of custom is_uri